### PR TITLE
dev-util/spirv-llvm-translator and dev-libs/opencl-clang: bump to v12.0.0

### PR DESCRIPTION
**Pull request by dAgbeve**
Package-Manager: Portage-3.0.20, Repoman-3.0.3
Signed-off-by: Douglas Agbeve <douglas@agbeve.com>

### CODE REVIEW

**Review by juippis — COMMENTED** [review]
I can't seem to compile these with llvm-12. Do you perhaps still have 11 installed?

```
CMake Error at CMakeLists.txt:35 (find_package):
  Could not find a configuration file for package "LLVM" that is compatible
  with requested version "11.1.0".

  The following configuration files were considered but not accepted:

    /usr/lib/llvm/12/lib/cmake/llvm/LLVMConfig.cmake, version: 12.0.1



-- Configuring incomplete, errors occurred!
See also "/var/tmp/portage/dev-util/spirv-llvm-translator-12.0.0/work/spirv-llvm-translator-12.0.0_build-abi_x86_32.x86/CMakeFiles/CMakeOutput.log".
 * ERROR: dev-util/spirv-llvm-translator-12.0.0::gentoo failed (configure phase):
 *   cmake failed
```

**Comment by dAgbeve** [comment]
It should compile fine now. Did not change the `src_url` to point to the right location, hence pulling in the older version files.

Also, the tests passed according to the following, so i enabled it.

`PASS: LLVM_SPIRV :: transcoding/unreachable.ll (504 of 510)`
`PASS: LLVM_SPIRV :: transcoding/vector_casts.ll (505 of 510)`
`PASS: LLVM_SPIRV :: vector_times_matrix.spt (506 of 510)`
`PASS: LLVM_SPIRV :: transcoding/vec_type_hint.cl (507 of 510)`
`PASS: LLVM_SPIRV :: vector_times_scalar.spt (508 of 510)`
`PASS: LLVM_SPIRV :: uitofp-with-bool.ll (509 of 510)`
`PASS: LLVM_SPIRV :: long-spec-const-composite.ll (510 of 510)`

`Testing Time: 10.61s`
  `Unsupported      :  39`
  `Passed           : 468`
 ` Expectedly Failed:   3`
`>>> Completed testing dev-util/spirv-llvm-translator-12.0.0`

**Review by juippis — COMMENTED** [review]
Can't install opencl-clang, which is a bit weird,
```
>>> Source configured.
>>> Compiling source in /var/tmp/portage/dev-libs/opencl-clang-12.0.0/work/opencl-clang-12.0.0 ...
 * abi_x86_32.x86: running multilib-minimal_abi_src_compile
 * Working in BUILD_DIR: "/var/tmp/portage/dev-libs/opencl-clang-12.0.0/work/opencl-clang-12.0.0_build-abi_x86_32.x86"
ninja -v -j16 -l10
ninja: error: 'llvm-tblgen', needed by 'opencl_clang_options.inc', missing and no known rule to make it
 * ERROR: dev-libs/opencl-clang-12.0.0::gentoo failed (compile phase):
 *   ninja -v -j16 -l10 failed
```
Not sure if the upstream cmake files needs patching or if this is a multilib issue. Since I do have the binary and can call it fine from command line.

**Comment by dAgbeve** [comment]
I dont seem to be able to replicate this error. Checking for the needed library is rightfully installed in `/usr/lib/llvm/12/bin/llvm-tblgen
` also it is only set in the `CMakeLists.txt`  as `set(LLVM_TABLEGEN_EXE "llvm-tblgen")` of the source file.

It installs on my side without error.

**Comment by juippis** [comment]
Yeah, it seems to be multilib-related. I can also install it fine if I turn off multilib. 

Wonder what to do here, I've got no clue how to fix it. There might be some examples in the tree available too
Crosschecking 
https://qa-reports.gentoo.org/output/eclass-usage/llvm.txt
and
https://qa-reports.gentoo.org/output/eclass-usage/cmake-multilib.txt
for common occurences is probably a good start...

**Comment by dAgbeve** [comment]
Poking around, there is a `patch file` that `find_program(LLVM_TABLEGEN_EXE "llvm-tblgen")` instead of just `setting it`. Could you apply that to see if it solves it.  It will mean to have 32-bit versions of all the dependencies, if i were to go multilib way.

The patch is at `files/opencl-clang-10.0.0.1_find-llvm-tblgen.patch`

**Review by juippis — APPROVED** [review]
That seems to fix it for me, thanks! I'll add the patch file.

File: dev-libs/opencl-clang/opencl-clang-12.0.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+EAPI=7
+
+CMAKE_ECLASS=cmake
+
+inherit cmake-multilib llvm
+
+DESCRIPTION="OpenCL-oriented thin wrapper library around clang"
+HOMEPAGE="https://github.com/intel/opencl-clang"
+SRC_URI="https://github.com/intel/${PN}/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz"
+S="${WORKDIR}/${PN}-${PV}"
```
Comment (juippis) [inline dev-libs/opencl-clang/opencl-clang-12.0.0.ebuild:13]: Oh right, should've fixed this. This is redundant.

**Pull request closed without merging**